pandas/tests/frame/methods/test_cov_corr.py

```diff
@@ -334,9 +331,7 @@ def test_corrwith_with_objects(self, using_infer_string):
         df2["obj"] = "bar"
 
         if using_infer_string:
-            import pyarrow as pa
-
-            with pytest.raises(pa.lib.ArrowNotImplementedError, match="has no kernel"):
+            with pytest.raises(TypeError, match="Cannot perform reduction"):
```

Comment:
now that the `ArrowNotImplementedError` is no longer raised, I wonder whether there is any value having the `if using_infer_string:` block or whether just to have the two messages combined like we do elsewhere, i.e. `msg - msg1|msg2` (prefered syntax is `"|".join(...`) and then `match=msg`.

However, this also raises the question is the new message that users see, `TypeError: Cannot perform reduction 'mean' with string dtype` and better/worse than the previous message, `TypeError: Could not convert ['foofoofoofoofoofoofoofoofoofoo'] to numeric`. As the most informative message would probably suggest that they use `numeric_only=True`?

Reply:
I think in this specific case it is useful to keep the `if using_infer_string`/`else` blocks, because then when cleaning up the usage of that, we can remove the entire `else` block, and we don't keep testing the two error messages later on. 

>  As the most informative message would probably suggest that they use numeric_only=True?

That would indeed be nice, but that's something specifically for the aggregation at the DataFrame level (e.g. `df.mean()`) to catch and amend the error message (not something specifically for the string dtype / array to do). But definitely worth having a separate issue for that.